Project/Source/Components/ComponentTrail.cpp

```diff
@@ -0,0 +1,237 @@
+	glUseProgram(program);
+	//TODO ADD DELTATIME
+
+	ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+
+	Frustum* frustum = App->camera->GetActiveCamera()->GetFrustum();
+	float4x4* proj = &App->camera->GetProjectionMatrix();
+	float4x4* view = &App->camera->GetViewMatrix();
+	//float4x4 MiguelMatrix = float4x4::LookAt(transform->GetGlobalMatrix().RotatePart().Col(2), -frustum->Front(), transform->GetGlobalMatrix().RotatePart().Col(1), float3::unitY);
+	/*float4x4 newModelMatrix = transform->GetGlobalMatrix().LookAt(transform->GetGlobalMatrix().RotatePart().Col(2), -frustum->Front(), transform->GetGlobalMatrix().RotatePart().Col(1), float3::unitY);
+	newModelMatrix = transform->GetGlobalMatrix();
+	float4x4 Final = float4x4::FromTRS(float3::zero, transform->GetGlobalMatrix().RotatePart(), transform->GetGlobalScale());*/
+
+	//-> glUniformMatrix4fv(glGetUniformLocation(program, "model"), 1, GL_TRUE, newModelMatrix.ptr());
+	float3 PolLocalMiuse = transform->GetRotation().ToEulerXYZ();
+	float3 PolMiuse = transform->GetGlobalRotation().ToEulerXYZ();
+	/*transform->SetRotation(float3(PolLocalMiuse.x, PolLocalMiuse.y, PolMiuse.y - (pi / 2)));*/
+
+	float4x4 newModelMatrix = float4x4::FromTRS(transform->GetGlobalPosition(), Quat::FromEulerXYZ(PolLocalMiuse.x, PolLocalMiuse.y, PolMiuse.y - (pi / 2)), transform->GetGlobalScale());
+
+	glActiveTexture(GL_TEXTURE0);
+	glUniformMatrix4fv(glGetUniformLocation(program, "model"), 1, GL_TRUE, newModelMatrix.ptr());
+	glUniformMatrix4fv(glGetUniformLocation(program, "view"), 1, GL_TRUE, view->ptr());
+	glUniformMatrix4fv(glGetUniformLocation(program, "proj"), 1, GL_TRUE, proj->ptr());
+	
+	//glUniform1i(glGetUniformLocation(program, "diffuse"), 0);
+	//glUniform1f(glGetUniformLocation(program, "currentFrame"), currentParticle.currentFrame);
+	//glUniform1f(glGetUniformLocation(program, "colorFrame"), currentParticle.colorFrame);
+	glUniform4fv(glGetUniformLocation(program, "inputColor"), 1, initC.ptr());
+	//glUniform4fv(glGetUniformLocation(program, "finalColor"), 1, finalC.ptr());
+
+	ResourceTexture* textureResource = App->resources->GetResource<ResourceTexture>(textureID);
+	if (textureResource != nullptr) {
+		glBindTexture(GL_TEXTURE_2D, textureResource->glTexture);
+	}
+
+	glDrawArrays(GL_TRIANGLES, 0, trianglesCreated);
+	glBindTexture(GL_TEXTURE_2D, 0);
+	glBindBuffer(GL_ARRAY_BUFFER, 0);
+	glEnable(GL_CULL_FACE);
+	glDisable(GL_BLEND);
+	glDepthMask(GL_TRUE);
+}
+
+void ComponentTrail::SpawnParticle() {
+}
+
+void ComponentTrail::UpdateVerticesPosition() {
+	for (size_t i = 0; i < (maxVertices - 30); i++) {
+		verticesPosition[i] = verticesPosition[i + 30];
+	}
+}
+void ComponentTrail::insertVertex(float3 vertex) {
+	verticesPosition[trianglesCreated++] = vertex.x;
+	verticesPosition[trianglesCreated++] = vertex.y;
+	verticesPosition[trianglesCreated++] = vertex.z;
+}
+
+void ComponentTrail::insertTextureCoords() {
```

Comment:
Make sure all the variable and function names are following the code style guide.